pkg/app/pipedv1/plugin/kubernetes/provider/diff.go

```diff
@@ -81,3 +85,143 @@ func normalizeNewSecret(old, new *unstructured.Unstructured) (*unstructured.Unst
 
 	return &unstructured.Unstructured{Object: newO}, nil
 }
+
+type DiffListResult struct {
+	Adds    []Manifest
+	Deletes []Manifest
+	Changes []DiffListChange
+}
+
+type DiffListChange struct {
+	Old  Manifest
+	New  Manifest
+	Diff *diff.Result
+}
+
+func (r *DiffListResult) NoChanges() bool {
+	return len(r.Adds)+len(r.Deletes)+len(r.Changes) == 0
+}
+
+func (r *DiffListResult) TotalOutOfSync() int {
+	return len(r.Adds) + len(r.Deletes) + len(r.Changes)
+}
+
+func DiffList(liveManifests, desiredManifests []Manifest, logger *zap.Logger, opts ...diff.Option) (*DiffListResult, error) {
+	adds, deletes, newChanges, oldChanges := groupManifests(liveManifests, desiredManifests)
+	result := &DiffListResult{
+		Adds:    adds,
+		Deletes: deletes,
+		Changes: make([]DiffListChange, 0, len(newChanges)),
+	}
+
+	for i := 0; i < len(newChanges); i++ {
+		diffResult, err := Diff(oldChanges[i], newChanges[i], logger, opts...)
+		if err != nil {
+			logger.Error("Failed to diff manifests", zap.Error(err))
+			continue
+		}
+		if !diffResult.HasDiff() {
+			continue
+		}
+		result.Changes = append(result.Changes, DiffListChange{
+			Old:  oldChanges[i],
+			New:  newChanges[i],
+			Diff: diffResult,
+		})
+	}
+
+	return result, nil
+}
+
+func groupManifests(olds, news []Manifest) (adds, deletes, newChanges, oldChanges []Manifest) {
```

Comment:
ref; https://github.com/pipe-cd/pipecd/blob/20577c01513594cf6d95e5d163a4925f066085b2/pkg/app/piped/platformprovider/kubernetes/diff.go#L266-L305

I changed it to compare with the namespace because the plugin's implementation sets the namespace when loading and parsing manifests, so I think it's ok.